addons/io_hubs_addon/components/definitions/media_frame.py

```diff
@@ -0,0 +1,60 @@
+from bpy.props import EnumProperty, FloatVectorProperty, BoolProperty
+from bpy.types import Node
+from ..hubs_component import HubsComponent
+from ..types import Category, PanelType
+from mathutils import Matrix
+
+
+class MediaFrame(HubsComponent):
+    _definition = {
+        'name': 'media-frame',
+        'display_name': 'Media Frame',
+        'category': Category.ELEMENTS,
+        'node_type': Node,
+        'panel_type': PanelType.OBJECT,
+        'icon': 'OBJECT_DATA',
+        'deps': ['networked']
+    }
+
+    bounds: FloatVectorProperty(
+        name="Bounds", description="Bounding box to fit objects into when they are snapped into the media frame.", unit='LENGTH', subtype="XYZ_LENGTH", default=(1.0, 1.0, 1.0))
+
+    mediaType: EnumProperty(
+        name="Media Type",
+        description="Limit what type of media this frame will capture",
+        items=[("all", "All Media", "Allow any type of media."),
+               ("all-2d", "Only 2D Media", "Allow only Images, Videos, and PDFs."),
+               ("model", "Only 3D Models", "Allow only 3D models."),
+               ("image", "Only Images", "Allow only images."),
+               ("video", "Only Videos", "Allow only videos."),
+               ("pdf", "Only PDFs", "Allow only PDFs.")],
+        default="all-2d")
+
+    snapToCenter: BoolProperty(
+        name="Snap To Center",
+        description="Snap the media to the center of the media frame when capturing. If set to false the object will just remain in the place it was dorpped but still be considered \"captured\" by the media frame.",
+        default=True
+    )
+
+    @classmethod
+    def create_gizmo(cls, obj, gizmo_group):
+        widget = gizmo_group.gizmos.new('GIZMO_GT_cage_3d')
+        widget.draw_style = ('BOX')
+        loc, rot, _ = obj.matrix_world.decompose()
+        widget.matrix_basis = Matrix.LocRotScale(loc, rot, obj.dimensions)
```

Comment:
Applogize if I just keep missing it, but I am still confused how the `bounds: FloatVectorProperty` is being applied to this gizmo. Ultimately that is what gets exported into the GLTF and what gets used at runtime to decide the bounds of the media frame.

In Spoke for media frame we don't have a "bounds" property and instead just have the user scale the media frame. Then at export time we clear out the scale of the object and set the bounds property based on scale. 

We could do that here too, but I think it would also be good to be able to just display/mutate a property with gizmos (I can definitely imagine components that might have multiple properties and cared about its object's scale too, so we couldn't really use this trick.

Reply:
Yeah, initially for some components I was just copying the preexisting behavior. I've updated the media-frame component to link the bounds property with the actual object dimension.

By the way while testing this I've seen that for media frame  we are not using the actual object dimension for the collider:
https://github.com/mozilla/hubs/blob/f1213d3e8b8a21960f49d1e7f0504825f59ceef8/src/gltf-component-mappings.js#L131-L138
It seems that we don't account for the objects transforms when creating the shape unless its FIT type is ALL.

I'll check that out later.